kyuubi-server/src/main/scala/org/apache/kyuubi/credentials/HadoopCredentialsManager.scala

```diff
@@ -146,12 +146,13 @@ class HadoopCredentialsManager private (name: String) extends AbstractService(na
   def sendCredentialsIfNeeded(
       sessionId: String,
       appUser: String,
-      send: String => Unit): Unit = {
+      send: String => Unit,
+      waitComplete: Boolean = false): Unit = {
```

Comment:
how about rename `waitComplete` to `onetime`?

In my opinion,

we want to enforce send credentials to engine side one time.
If it fails, we do not need wait it retry and just provide some prompt(the exception message) for client side.

Reply:
cc @zhouyifan279

Reply:
updated